### ISSUE TITLE
jupyterlab from conda-forge seg faults on mac

### ISSUE THREAD

**Issue by DanielGoldfarb** [jupyterlab/jupyterlab#14495]
## Description

- On macOS, conda-forge version of jupyterlab=3 seg faults when running `jupyter subcommand`
- (works fine on my WSL2 Ubuntu machine, but not on my macbook; see details below)

## Reproduce
- I am running macOS 10.13.6 
1.  `conda create -n testenv -c conda-forge -c nodefaults jupyterlab=3` 
2. `conda activate testenv`
3. `jupyter labextension list` or `jupyter lab`
  ```
  Segmentation fault: 11
  ```
## Context

- macOS 10.31.6
- using miniconda3 `conda --version` gives:  `conda 23.1.0`
- `jupyter --version`
  ```
  Selected Jupyter core packages...
  IPython          : 8.13.2
  ipykernel        : 6.22.0
  ipywidgets       : not installed
  jupyter_client   : 8.2.0
  jupyter_core     : 5.3.0
  jupyter_server   : 2.5.0
  jupyterlab       : 3.5.3
  nbclient         : 0.7.4
  nbconvert        : 7.3.1
  nbformat         : 5.8.0
  notebook         : 6.5.4
  qtconsole        : not installed
  traitlets        : 5.9.0
  ```
### Additional Details

- works fine on my WSL2 Ubuntu (only having problems on my mac)
- if I create a conda environment *without* jupyterlab, then activate, and then `pip install jupyterlab` then it works ok in that conda environment.  (implies that the conda-forge build has a problem on the mac)
- running `jupyter` with no args, or with no subcommand, works fine.  For example:
  ```
  jupyter <enter>
  jupyter --help <enter>
  jupyter --version <enter>  # all of these work fine.
  ```
- running `jupyter` with any **subcommand**, gives `Segmentation fault: 11`, for example:
  ```
  jupyter labextension list <enter>
  jupyter lab <enter>
  jupyter notebook <enter>  # All of these Segmentation fault: 11
  ``` 
- The Seg fault is always ***while importing main for the subcommand***, for example:
  - "jupyter notebook <enter>" crashes on: 
    ```
    from notebook.notebookapp import main
    ```
  - "jupyter labextension list <enter>" crashes on:
    ```
    from jupyterlab.labextensions import main
    ```

- If I had another macbook, I would test if this is an issue with all macs, or just mine.  As mentioned above, I don't have this problem on WSL2 ubuntu.  That said, it appears that I am not the only one experiencing this issue; see also:
  - https://stackoverflow.com/questions/75994426/  <-- Someone else with similar issue
  - https://stackoverflow.com/questions/76124633/  <-- This is me

- pdb seemed to indicate that the crash was in `lib/python3.11/importlib/_bootstrap.py` but that does not mean that the problem isn't with the jupyter package (from conda) only that (as I mentioned above) the crash occurs *while importing main from the specified subcommand*.

**Comment by fcollonval** [comment]
Thanks for reporting @DanielGoldfarb

would you mind providing the processor type? Is it a ARM processor?

> if this is the case, I guess the seg fault is related to some Rust dependencies.

Could you also try the up coming 4.0.0 version (we are at release candidate phase)? You can try it using:

```
pip install --pre jupyterlab
```

**Comment by fcollonval** [comment]
or with conda:

```
conda install -c "conda-forge/label/jupyterlab_rc" jupyterlab
```

**Comment by DanielGoldfarb** [comment]
Processoris 2.5 GHz Intel Core i7

I forgot to mention above, I have two conda environments from towards the beginning of the month of April where the conda install of jupyterlab install worked fine.  This problem appears to have started approximately mid-April.

I will try the release candidate later today when I have more time.

**Comment by fcollonval** [comment]
Thanks @DanielGoldfarb - then I guess this is not related to the rust packages. If you could list the package installed in the two environments (the one working and the one broken), the root cause is probably in one of the packages that got updated.

**Comment by bollwyvl** [comment]
Indeed, `conda list --explicit` from the two environments would be the ideal artifact....

However, I'd recommend posting that as a new issue on https://github.com/conda-forge/jupyterlab-feedstock... while there is some overlap between the two sets of maintainers, _this_ repo's primary job is to build PyPI distributions, and can really only do its best effort with the official Python distributions on CI.

**Comment by DanielGoldfarb** [comment]
@bollwyvl  
makes sense.  will do, but probably won't get to it until next week (hopefully early next week).  

at present I have a work-around: after creating a conda environment without jupyterlab, i then install jupyterlab from pypi instead of from conda-forge.

**Comment by fcollonval** [comment]
A workaround has been found by @marcbernot (:heart:) as reported [there](https://github.com/conda-forge/jupyterlab-feedstock/issues/378#issuecomment-1572731937)

> Hello I got the same issue with macOS Mojave 10.14.6. Since I still have some not updated working environments I was able to track this down to a change of libffi version (and other related differences).
> 
> You can recover a working environment with last jupyterlab version by pinning libffi to 3.3. `micromamba create -f https://prefix.dev/envs/marcbernot/mojave/conda-lock.yml -n mojave` or `conda create -n testmojave python=3.9 libffi=3.3 notebook jupyterlab`
> 
> If someone wants to investigate further, these are two environments with python=3.9.7. The first one is ok (libffi=3.3), the second one gives the segmentation fault (libffi>3.3).
> 
> ```
> micromamba create -f https://prefix.dev/envs/marcbernot/mojave/conda-lock.yml -n mojave
> micromamba create -f https://prefix.dev/envs/marcbernot/mojave_nok/conda-lock.yml -n mojave_nok
> ```
> 
> A `conda compare` of the two environments yields
> 
> ```
> cffi found but mismatch. Specification pkg: cffi==1.15.1=py39h131948b_3, Running pkg: cffi==1.14.6=py39hb71fe58_0
> libffi found but mismatch. Specification pkg: libffi==3.4.2=h0d85af4_5, Running pkg: libffi==3.3=h046ec9c_2
> openssl found but mismatch. Specification pkg: openssl==3.1.1=h8a1eda9_1, Running pkg: openssl==1.1.1u=h8a1eda9_0
> pyobjc-core found but mismatch. Specification pkg: pyobjc-core==9.1.1=py39h0504f80_0, Running pkg: pyobjc-core==7.3=py39h89e85a6_2
> pyobjc-framework-cocoa found but mismatch. Specification pkg: pyobjc-framework-cocoa==9.1.1=py39h0504f80_0, Running pkg: pyobjc-framework-cocoa==7.3=py39h6e9494a_1
> python found but mismatch. Specification pkg: python==3.9.7=h38b4d05_3_cpython, Running pkg: python==3.9.7=hd187cdc_0_cpython
> ```

**Comment by MMesch** [comment]
The triaging agent has identified this issue as likely resolved. Its rationale was a bit off so I'm not posting it. But reading through the thread I agree that it can be closed. The most useful thread to look at if you encounter the problem is probably https://github.com/conda-forge/jupyterlab-feedstock/issues/378#issuecomment-1572731937 . Please reopen if you think anything remains to be done in this repo and comment what it is.
